content/en/serverless/guide/step_functions_cdk.md

```diff
@@ -0,0 +1,240 @@
+---
+title: CDK Examples for Instrumenting AWS Step Functions
+further_reading:
+    - link: "/serverless/step_functions/installation"
+      tag: "Documentation"
+      text: "Install Serverless Monitoring for AWS Step Functions"
+---
+
+This page lists language-specific code examples for [instrumenting AWS Step Functions][1] using [Datadog's CDK Construct Library][2].
+
+## Basic setup
+
+{{< tabs >}}
+{{% tab "TypeScript" %}}
+
+Example stack: [`step-functions-typescript-stack`][1]
+
+{{< code-block lang="typescript" disable_copy="false" >}}
+import * as sfn from "aws-cdk-lib/aws-stepfunctions";
+import { DatadogStepFunctions} from "datadog-cdk-constructs-v2";
+
+const stateMachine = new sfn.StateMachine(...);
+const datadogSfn = new DatadogStepFunctions(this, "DatadogSfn", {
+  env: "<ENV>", // e.g. "dev"
+  service: "<SERVICE>", // e.g. "my-cdk-service"
+  version: "<VERSION>", // e.g. "1.0.0"
+  forwarderArn: "<FORWARDER_ARN>", // e.g. "arn:test:forwarder:sa-east-1:12345678:1"
+  tags: <TAGS>, // optional, e.g. "custom-tag-1:tag-value-1,custom-tag-2:tag-value-2"
+});
+datadogSfn.addStateMachines([stateMachine]);
+{{< /code-block >}}
+
+[1]: https://github.com/DataDog/datadog-cdk-constructs/tree/main/examples/step-functions-typescript-stack
+
+{{% /tab %}}
+{{% tab "Python" %}}
+
+Example stack: [`step-functions-python-stack`][1]
+
+{{< code-block lang="python" disable_copy="false" >}}
+from aws_cdk import (
+    aws_stepfunctions as sfn,
+    aws_stepfunctions_tasks as tasks,
+)
+from datadog_cdk_constructs_v2 import DatadogStepFunctions, DatadogLambda
+
+state_machine = sfn.StateMachine(...)
+datadog_sfn = DatadogStepFunctions(
+    self,
+    "DatadogSfn",
+    env="<ENV>", # e.g. "dev"
+    service="<SERVICE>", # e.g. "my-cdk-service"
+    version="<VERSION>", # e.g. "1.0.0"
+    forwarderArn="<FORWARDER_ARN>", # e.g. "arn:test:forwarder:sa-east-1:12345678:1"
+    tags=<TAGS>, # optional, e.g. "custom-tag-1:tag-value-1,custom-tag-2:tag-value-2"
+)
+datadog_sfn.add_state_machines([child_state_machine, parent_state_machine])
+{{< /code-block >}}
+
+[1]: https://github.com/DataDog/datadog-cdk-constructs/tree/main/examples/step-functions-python-stack
+{{% /tab %}}
+{{% tab "Go" %}}
```

Comment:
Just a question, but do we only have the official CDK constructs available in Go, TS and Python? And no support for .NET/Java right?

Reply:
.NET & Java definitely aren't as widely used, I'm just thinking for completeness that's all. So we can say we support all runtimes.

Is there not a way with the CDK to write it once in Javascript and then auto-gen the other runtimes though?

Reply:
This is a CDK construct for defining .NET Lambda functions. The construct is defined once in Typescript and then uses Projen to generate all the other packages. Could we not do the same for our constructs?

https://github.com/cdklabs/awscdk-lambda-dotnet/blob/main/.github/workflows/release.yml

Reply:
Yes, I think it's 1–2 week one-time work to support a new language.
From what I understand, we often using a "lazy" approach that we don't build something unless some customer asks for it. Have you heard of any customer ask, or could you align with Sumedha on this?